micropip/manager.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, List
+
+from micropip import package_index
+
+
+class Manager:
+    """
+    Manager provides an extensible interface for customizing micropip's behavior.
+
+    Each Manager instance holds its own local state that is
+    independent of other instances, including the global state.
+
+    TODO: Implement all of the following global commands to utilize local state.
+    """
+
+    def __init__(self):
+        self.index_urls = package_index.DEFAULT_INDEX_URLS
+
+        # TODO: initialize the compatibility layer
+        self.repodata_packages: dict[str, dict[str, Any]] = {}
+        self.repodata_info: dict[str, str] = {}
+
+        pass
+
+    def install(self):
+        pass
+
+    def list(self):
+        pass
+
+    def freeze(self) -> str:
+        pass
+
+    def add_mock_package(self):
+        pass
+
+    def list_mock_packages(self):
+        pass
+
+    def remove_mock_package(self):
+        pass
+
+    def uninstall(self):
+        pass
+
+    def set_index_urls(self, urls: List[str] | str):
```

Comment:
This was because of the name conflict with the `list()` method, unlike the `_list` export from `_commands.list.py` I couldn't alias the method name to something else. My LSP was being a bit weird about using `list` here but I suppose its a local issue..

Reply:
I see, that's funny. It was probably a bad decision to call it `list` at the first time. Anyway, it's not an issue. So let's keep this as-is.

Reply:
Add a comment next to the `from typing import List` please!